### ISSUE TITLE
Unprivileged container fails to start on a SYSV running host

### ISSUE THREAD

**Issue by pmaziere** [lxc/lxc#1279]
Hello,

I use lxc 2.0.5 on debian testing **without** systemd, and decided to move my containers from privileged to unprivileged, but could not succeed. 

So I started from scratch, and after several attempts, I discovered that a freshly created container (as non-root user `lxc-create -t download`) refuses to start, whereas if I replace sysv by systemd (`apt-get autoremove --purge sysvinit-core`) and reboot, it starts as expected. 

Reinstalling sysv (`apt-get install sysvinit-core`) and rebooting flips back to a non starting container.

Is sysv not supported for unprivileged containers or is this a bug ?

Here is the log produced by lxc-start run as a non-root user on a host without sysv:
```
      lxc-start 20161105235329.097 INFO     lxc_start_ui - tools/lxc_start.c:main:264 - using rcfile /home/pierre/.local/share/lxc/testbase/config
      lxc-start 20161105235329.105 WARN     lxc_confile - confile.c:config_pivotdir:1879 - lxc.pivotdir is ignored.  It will soon become an error.
      lxc-start 20161105235329.139 INFO     lxc_confile - confile.c:config_idmap:1500 - read uid map: type u nsid 0 hostid 100000 range 65537
      lxc-start 20161105235329.139 INFO     lxc_confile - confile.c:config_idmap:1500 - read uid map: type g nsid 0 hostid 100000 range 65537
      lxc-start 20161105235329.140 INFO     lxc_start - start.c:lxc_check_inherited:252 - closed inherited fd 4
      lxc-start 20161105235329.143 INFO     lxc_container - lxccontainer.c:do_lxcapi_start:797 - Attempting to set proc title to [lxc monitor] /home/pierre/.local/share/lxc testbase
      lxc-start 20161105235329.143 INFO     lxc_lsm - lsm/lsm.c:lsm_init:48 - LSM security driver AppArmor
      lxc-start 20161105235329.144 INFO     lxc_start - start.c:lxc_check_inherited:252 - closed inherited fd 4
      lxc-start 20161105235329.146 INFO     lxc_monitor - monitor.c:lxc_monitor_sock_name:178 - using monitor sock name lxc/717cadb10e1fbd77//home/pierre/.local/share/lxc
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:411 - processing: .reject_force_umount  # comment this to allow umount -f;  not recommended.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:576 - Adding native rule for reject_force_umount action 0.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:do_resolve_add_rule:251 - Setting Seccomp rule to reject force umounts.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:579 - Adding compat rule for reject_force_umount action 0.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:do_resolve_add_rule:251 - Setting Seccomp rule to reject force umounts.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:411 - processing: .[all].
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:411 - processing: .kexec_load errno 1.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:576 - Adding native rule for kexec_load action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:579 - Adding compat rule for kexec_load action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:411 - processing: .open_by_handle_at errno 1.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:576 - Adding native rule for open_by_handle_at action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:579 - Adding compat rule for open_by_handle_at action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:411 - processing: .init_module errno 1.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:576 - Adding native rule for init_module action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:579 - Adding compat rule for init_module action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:411 - processing: .finit_module errno 1.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:576 - Adding native rule for finit_module action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:579 - Adding compat rule for finit_module action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:411 - processing: .delete_module errno 1.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:576 - Adding native rule for delete_module action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:579 - Adding compat rule for delete_module action 327681.
      lxc-start 20161105235329.151 INFO     lxc_seccomp - seccomp.c:parse_config_v2:589 - Merging in the compat Seccomp ctx into the main one.
      lxc-start 20161105235329.151 DEBUG    lxc_start - start.c:setup_signal_fd:290 - sigchild handler set
      lxc-start 20161105235329.151 DEBUG    console - console.c:lxc_console_peer_default:469 - no console peer
      lxc-start 20161105235329.254 INFO     lxc_start - start.c:lxc_init:489 - 'testbase' is initialized
      lxc-start 20161105235329.255 DEBUG    lxc_start - start.c:__lxc_start:1327 - Not dropping cap_sys_boot or watching utmp
      lxc-start 20161105235329.255 INFO     lxc_start - start.c:resolve_clone_flags:1014 - Cloning a new user namespace
      lxc-start 20161105235329.255 INFO     lxc_cgroup - cgroups/cgroup.c:cgroup_init:68 - cgroup driver cgroupfs initing for testbase
      lxc-start 20161105235329.255 ERROR    lxc_cgfs - cgroups/cgfs.c:lxc_cgroupfs_create:889 - Could not find writable mount point for cgroup hierarchy 12 while trying to create cgroup.
      lxc-start 20161105235329.256 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/pids/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.256 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/perf_event/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.256 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/net_prio/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.256 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/net_cls/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.257 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/memory//lxc
      lxc-start 20161105235329.258 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/memory//user/pierre/0
      lxc-start 20161105235329.258 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/memory//user/pierre
      lxc-start 20161105235329.258 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/memory//user
      lxc-start 20161105235329.258 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/memory//user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.259 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/memory//user.slice/user-1000.slice
      lxc-start 20161105235329.259 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/memory//user.slice
      lxc-start 20161105235329.259 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/memory/
      lxc-start 20161105235329.259 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/freezer//lxc
      lxc-start 20161105235329.259 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/freezer//user/pierre/0
      lxc-start 20161105235329.259 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/freezer//user/pierre
      lxc-start 20161105235329.259 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/freezer//user
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/freezer//user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/freezer//user.slice/user-1000.slice
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/freezer//user.slice
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/freezer/
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/devices/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/cpuset/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/cpuacct/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/cpu/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.260 ERROR    lxc_cgfs - cgroups/cgfs.c:cgroup_rmdir:209 - Permission denied - cgroup_rmdir: failed to delete /sys/fs/cgroup/blkio/user.slice/user-1000.slice/session-1.scope
      lxc-start 20161105235329.260 ERROR    lxc_start - start.c:lxc_spawn:1101 - failed creating cgroups
      lxc-start 20161105235329.260 ERROR    lxc_start - start.c:__lxc_start:1354 - failed to spawn 'testbase'
      lxc-start 20161105235329.260 INFO     lxc_conf - conf.c:run_script_argv:367 - Executing script '/usr/share/lxcfs/lxc.reboot.hook' for container 'testbase', config section 'lxc'
      lxc-start 20161105235329.782 WARN     lxc_commands - commands.c:lxc_cmd_rsp_recv:172 - command get_cgroup failed to receive response
      lxc-start 20161105235334.786 ERROR    lxc_start_ui - tools/lxc_start.c:main:344 - The container failed to start.
      lxc-start 20161105235334.786 ERROR    lxc_start_ui - tools/lxc_start.c:main:346 - To get more details, run the container in foreground mode.
      lxc-start 20161105235334.786 ERROR    lxc_start_ui - tools/lxc_start.c:main:348 - Additional information can be obtained by setting the --logfile and --logpriority options.
```

thanks for your inputs

**Comment by stgraber** [comment]
```
      lxc-start 20161105235329.255 ERROR    lxc_cgfs - cgroups/cgfs.c:lxc_cgroupfs_create:889 - Could not find writable mount point for cgroup hierarchy 12 while trying to create cgroup.
```

Sounds like you're missing a cleanly setup /sys/fs/cgroup/ tree so LXC can't setup the container.

You probably want to install cgroupfs-mount and libpam-cgfs, then reboot. The combination of those two should get you all the right cgroups mounted properly AND make the cgroups writable for your user at login time.

**Comment by pmaziere** [comment]
Both cgroupfs-mount and libpam-cgfs are already installed.
But cgroupfs files which have the writable permission are only writable for root.
When I check the pam configuration I can see this in /etc/pam.d/common-session
```
session	optional	pam_cgfs.so -c freezer,memory,name=systemd
```

having a systemd reference here could be an issue on a host without systemd, couldn't it ?
Isn't it similar to what was implied by @brauner at the end of the thread of #1258 ?

**Comment by evgeni** [comment]
Can you post `cat /proc/self/cgroup` from the shell you are trying to start the container from?

**Comment by pmaziere** [comment]
`cat /proc/self/cgroup`
```
12:name=systemd:/user.slice/user-1000.slice/session-1.scope
11:pids:/user.slice/user-1000.slice/session-1.scope
10:perf_event:/user.slice/user-1000.slice/session-1.scope
9:net_prio:/user.slice/user-1000.slice/session-1.scope
8:net_cls:/user.slice/user-1000.slice/session-1.scope
7:memory:/
6:freezer:/
5:devices:/user.slice/user-1000.slice/session-1.scope
4:cpuset:/user.slice/user-1000.slice/session-1.scope
3:cpuacct:/user.slice/user-1000.slice/session-1.scope
2:cpu:/user.slice/user-1000.slice/session-1.scope
1:blkio:/user.slice/user-1000.slice/session-1.scope
```

**Comment by evgeni** [comment]
@pmaziere that is with systemd or with sysvinit?

**Comment by pmaziere** [comment]
with syvinit:

`dpkg -l *systemd*`
```
ii  dh-systemd                                10.2.2                    all    
ii  libpam-systemd:amd64                      231-9                     amd64  
ii  libsystemd-login0:amd64                   215-18                    amd64  
ii  libsystemd0:amd64                         231-9                     amd64  
ii  libsystemd0:i386                          231-9                     i386   
ii  systemd                                   231-9                     amd64  
un  systemd-container                         <none>                    <none> 
ii  systemd-shim                              10-2                      amd64  
un  systemd-sysv                              <none>                    <none> 
un  systemd-ui                                <none>                    <none>
```

`dpkg -l sysv*`
```
ii  sysv-rc                                   2.88dsf-59.8              all  
ii  sysvinit                                  2.88dsf-59.7              amd64
ii  sysvinit-core                             2.88dsf-59.8              amd64
ii  sysvinit-utils                            2.88dsf-59.8              amd64
```

**Comment by brauner** [comment]
Ok, so you clearly have a busted init situation. Both init systems, `systemd` and `sysvinit` are installed. The `ii` in the `dpkg -l` output means that the package is currently installed. And from the output in `cat /proc/self/cgroup` it also seems obvious that you're currently running `systemd` as the `name=systemd` controller is mounted.  The solution is to:

- decide for an init system and remove *any references to the other init system`
- make sure that in case you're deciding to tun a `sysvinit` init system, to install the packages that @stgraber mentioned
- *reboot* your system to end up with a correctly mounted cgroup system

If the problem still persists after that, please re-open.

**Comment by pmaziere** [comment]
I did all this except for removing libsystemd0 required by so many other packages.

The reason I did not do it previously was because removing systemd related packages failed to launch Xorg with `startx`. 
This forced me to look further into this issue, and today I discovered the xserver-xorg-legacy package and the `needs_root_rights = yes` option to set in /etc/X11/Xwrapper.config. This makes Xorg works again for a systemd-almost-free debian, in case it can be of any use for someone else.

Now, I can not start a jessie container because lxc fails to start a systemd enabled container:
`Failed to mount cgroup at /sys/fs/cgroup/systemd: Operation not permitted`

but I created a wheezy container and it works like a charm. Maybe upgrading from there to jessie and removing systemd can be the solution.

Anyway that is another story.
Thanks for pushing me in the right direction.
